_posts/2021-07-05-newsletter.md

```diff
@@ -0,0 +1,199 @@
+- [Architext](https://github.com/vigoux/architext.nvim) -- structural text editing (i.e., syntax-aware search and replace).
+
+More information on Neovim's tree-sitter features can be found in [Nvim-treesitter](https://github.com/nvim-treesitter/nvim-treesitter) and read [`:h treesitter`](https://neovim.io/doc/user/treesitter.html); you should also check out Thomas Vigouroux's [Vimconf.live presentation](https://youtu.be/EReNOGuMBmo?t=515). 
+
+Although these plugins already work reasonably well for some languages and features, there are still a number of bugs and performance regressions that may occur regularly and that need to be addressed before tree-sitter support in Neovim can be declared stable. Note also that enabling tree-sitter based highlighting for a language currently disables the internal regex-based syntax engine _completely_ for this file type (which may break other features that rely on it). Fixing these issues and improving the API (which is not stable yet) will be a major focus of the development cycle leading up to the 0.6 release.
+
+
+### What's the difference between LSP and tree-sitter?
+
+This is a common question, especially since LSP as of version 3.16 provides "semantic highlighting". In short, tree-sitter operates on a single file, parsing the file into a syntax tree which is used to support the variety of enhanced code navigation and manipulation functions. On the other hand, language servers operate across multiple files and project libraries, using various different, server-dependent, methods for parsing each file's syntax tree. (Of course, tree-sitter is one possible choice for this purpose, and is in fact used by, e.g., [bash-language-server](https://github.com/bash-lsp/bash-language-server) and [wasm-language-server](https://github.com/wasm-lsp/wasm-language-server).)
+
+In particular, this means that language servers can use _semantic_ information from a different file to annotate the tree for the current file: For example, a variable declared as `const` in one file can be highlighted in red if it is used in a _different_ file -- something that tree-sitter cannot do since it only has access to the latter file when highlighting.
+
+For more details, watch TJ DeVries' [presentation on this topic](https://www.youtube.com/watch?v=c17j09vY5sw).
+
+
+## Other changes
+
+Of course, these were not the only [major changes in 0.5](https://github.com/neovim/neovim/commit/a5ac2f45ff84a688a09479f357a9909d5b914294). Here is a short summary of representative new features.
+
+### Decorations
+
+There is an improved "decoration provider" API that allows setting and interacting with "extmarks" (invisible anchored text markers that move when surrounding text is edited), "virtual text" (text overlays that can now be drawn at any position on screen), and highlights. (This API is heavily leveraged by tree-sitter.)
```

Comment:
I didn't know I had an OCD before I'd seen your parentheses :p   `(This API is heavily leveraged by tree-sitter.)`

Reply:
You and me, both :)

Was this an actual suggestion, though, or just copied from the text? (Because I don't see a difference...)

Reply:
I would also remove the "quotes" around "virtual text" as that is existing terminology since quite some time :)

Reply:
But yeah, I can remove the parentheses with a slight reformulation.

Reply:
good call; no need for scare quotes.